pkg/codegen/schema/schema.go

```diff
@@ -611,17 +641,35 @@ type PropertySpec struct {
 type ObjectTypeSpec struct {
 	// Description is the description of the type, if any.
 	Description string `json:"description,omitempty"`
-	// Properties is a map from property name to PropertySpec that describes the type's properties.
+	// Properties, if present, is a map from property name to PropertySpec that describes the type's properties.
 	Properties map[string]PropertySpec `json:"properties,omitempty"`
-	// Type must be "object".
+	// Type must be "object" if this is an object type, or the underlying type for an enum.
 	Type string `json:"type,omitempty"`
-	// Requires is a list of the names of the type's required properties. These properties must be set for inputs and
-	// will always be set for outputs.
+	// Required, if present is a list of the names of an object type's required properties. These properties must be set
+	// for inputs and will always be set for outputs.
 	Required []string `json:"required,omitempty"`
 	// Language specifies additional language-specific data about the type.
 	Language map[string]json.RawMessage `json:"language,omitempty"`
 }
 
+// ComplexTypeSpec is the serializable form of an object or enum type.
+type ComplexTypeSpec struct {
+	ObjectTypeSpec
+
+	// Enum, if present, if the list of possible values for an enum type.
+	Enum []*EnumValueSpec `json:"enum,omitempty"`
```

Comment:
In this case, we should also require that the `Type` field is `"enum"`.

Reply:
The Type field for enums is usually used to capture the underlying primitive type, i.e. `string`. I was trying to keep this consistent with the [spec](https://swagger.io/docs/specification/data-models/enums/#ctxM:~:text=schema%3A,%20%20%20%20%20%20%20%20%20enum%3A%20%5Basc%2C%20desc%5D,-In). I flip-flopped on this a bunch, are you suggesting there should be an `ElementType` and a `Type` in the json schema?

Reply:
> I was trying to keep this consistent with the spec. 

Ah, perfect! That works, then.